src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineInstance.java

```diff
@@ -41,6 +42,7 @@ public ComputeEngineInstance(String cloudName,
                                  String nodeDescription,
                                  String sshUser,
                                  String remoteFS,
+                                 WindowsConfiguration windowsConfig,
```

Comment:
If this isn't required for non-windows scenarios, then this param should be made to be optional, see: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
Also the corresponding instance variable should be made to be the type: Optional<T>.

Reply:
done